mkosi/config.py

```diff
@@ -1010,12 +1008,12 @@ def config_match_version(match: str, value: str) -> bool:
 def config_make_dict_parser(
     *,
     delimiter: Optional[str] = None,
-    parse: Callable[[str], tuple[str, Any]],
+    parse: Callable[[str], tuple[str, str]],
     unescape: bool = False,
     allow_paths: bool = False,
     reset: bool = True,
-) -> ConfigParseCallback:
-    def config_parse_dict(value: Optional[str], old: Optional[dict[str, Any]]) -> Optional[dict[str, Any]]:
+) -> ConfigParseCallback[dict[str, str]]:
+    def config_parse_dict(value: Optional[str], old: Optional[dict[str, str]]) -> Optional[dict[str, str]]:
```

Comment:
The change from `Any` to `str` was done because otherwise the parser is unsound if `allow_paths=True`. If this should ever be required we would likely need to split `parse` into `splitter: str -> (key: str, val: str)` and `parser: (val: str) -> (val: T)`. This change only enforces it at the type level